### ISSUE TITLE
Improve evaluation results for Batch norm

### ISSUE THREAD

**Issue by astonzhang** [d2l-ai/d2l-en#2108]
http://preview.d2l.ai.s3-website-us-west-2.amazonaws.com/d2l-en/master/chapter_convolutional-modern/batch-norm.html

The evaluation results of batch norm for all frameworks need improvements. For example, as the text states, the performance should be better than that of [LeNet without BN](http://preview.d2l.ai.s3-website-us-west-2.amazonaws.com/d2l-en/master/chapter_convolutional-neural-networks/lenet.html). However, the training looks unstable as shown in the plots, such as in the plot of val_acc.

![Screen Shot 2022-04-20 at 6 06 31 PM](https://user-images.githubusercontent.com/22279212/164351366-112f54e6-d422-4a09-b00d-3d02168f2f90.png)

![Screen Shot 2022-04-20 at 6 06 47 PM](https://user-images.githubusercontent.com/22279212/164351396-ede07e06-dd13-49da-919f-dedbe70ea9c1.png)

**Comment by astonzhang** [comment]
@AnirudhDagar feel free to share your insight here while you're working on https://github.com/d2l-ai/d2l-en/issues/2107 and https://github.com/d2l-ai/d2l-en/issues/2099

**Comment by cheungdaven** [comment]
@astonzhang I found that removing the batchnorm on the fully-connected layers can make the curve stable.

**Comment by astonzhang** [comment]
@cheungdaven In fact, it seems that ResNet and DenseNet have similar issues:

http://preview.d2l.ai.s3-website-us-west-2.amazonaws.com/d2l-en/master/chapter_convolutional-modern/resnet.html
http://preview.d2l.ai.s3-website-us-west-2.amazonaws.com/d2l-en/master/chapter_convolutional-modern/densenet.html

although RegNet in PyTorch has smoother plots:
http://preview.d2l.ai.s3-website-us-west-2.amazonaws.com/d2l-en/master/chapter_convolutional-modern/cnn-design.html

Since ResNet and DenseNet do not apply BN on the FC layers in network heads, perhaps the plot issue is with somewhere else? Are you able to find any literature that supports removal of BN after FC layers? If not, could you try something else?

**Comment by AnirudhDagar** [comment]
Closing this since it was fixed with the latest version of all the frameworks.
